libs/wingsdk/src/target-sim/function.inflight.ts

```diff
@@ -9,63 +13,178 @@ import {
 import { TraceType } from "../std";
 
 export class Function implements IFunctionClient, ISimulatorResourceInstance {
-  private readonly filename: string;
+  private readonly originalFile: string;
+  private bundledFile: string | undefined;
   private readonly env: Record<string, string>;
   private readonly context: ISimulatorContext;
   private readonly timeout: number;
-  private readonly sandbox: Sandbox;
+  private readonly maxWorkers: number;
+  private readonly workers = new Array<Sandbox>();
+  private createBundlePromise: Promise<void>;
 
   constructor(props: FunctionSchema["props"], context: ISimulatorContext) {
     if (props.sourceCodeLanguage !== "javascript") {
       throw new Error("Only JavaScript is supported");
     }
-    this.filename = path.resolve(context.simdir, props.sourceCodeFile);
+    this.originalFile = path.resolve(context.simdir, props.sourceCodeFile);
     this.env = props.environmentVariables ?? {};
     this.context = context;
     this.timeout = props.timeout;
-    this.sandbox = new Sandbox(this.filename, {
-      env: {
-        ...this.env,
-        WING_SIMULATOR_URL: this.context.serverUrl,
-      },
-      timeout: this.timeout,
-      log: (internal, _level, message) => {
-        this.context.addTrace({
-          data: { message },
-          type: internal ? TraceType.RESOURCE : TraceType.LOG,
-          sourcePath: this.context.resourcePath,
-          sourceType: FUNCTION_FQN,
-          timestamp: new Date().toISOString(),
-        });
-      },
-    });
+    this.maxWorkers = props.concurrency;
+
+    this.createBundlePromise = this.createBundle();
+  }
+
+  // Create a bundle with the code that will be run by all workers.
+  private async createBundle() {
```

Comment:
We're now creating multiple sandboxes (one per worker), but we only need to do the code bundling once since all workers are executing the same. So I've moved this responsibility from the `Sandbox` class to the `FunctionClient` class.

Reply:
Kind of feels like this is not coupled with `cloud.Function` because it will be needed by anyone who uses the sandbox. Perhaps make this a static method of `Sandbox` (`Sandbox.createBundle()`)?

And make this function return something like `SandboxBundle` which will be the input to the sandbox. No?

Reply:
Yes that sounds like a good improvement